index.js

```diff
@@ -600,6 +598,30 @@ function hasEncryptionScheme(mediaKeySystemAccess) {
   return false;
 }
 
+/**
+ * @param {?string} scheme Encryption scheme to check
+ * @param {?string} supportedScheme A guess at the encryption scheme this
+ *   supports.
+ * @return {boolean} True if the scheme is compatible.
+ */
+function checkSupportedScheme(scheme, supportedScheme) {
+  if (scheme && scheme != supportedScheme) {
```

Comment:
I would find this logic more readable if we flattened it and had early-returns for each special case, one at a time.  For example:

```js
if (!scheme) {
  // Not encrypted = always supported
  return true;
}

if (scheme == supportedScheme) {
  // The assumed-supported legacy scheme for this platform.
  return true;
}

if (scheme == 'cbcs' || scheme == 'cbcs-1-9') {
  if (isRecentFirefox || isChromecast) {
    // Firefox >= 100 supports CBCS, but doesn't support queries yet.
    // Older Chromecast devices are assumed to support CBCS as well.
    return true;
  }
}

return false;
```

Reply:
Done!